### PR TITLE
Dash web trader

### PR DESCRIPTION
**Pull request by ronniebugia**
Issue for app: #45

# App pull request
- [ ] This is a new app
- [x] I am improving an existing app (redesigns/code "makeovers")

## About

- Playground deployment URL (new version): https://dash-playground.plotly.host/dash-web-trader/
- Current gallery app URL: https://dash-gallery.plotly.host/dash-web-trader/

## Workflow

- [x] I have created a branch in the appropriate monorepo, and the
  elements necessary for successful deployment are in place.
- [x] If the app is a redesigned and/or restyled version of an
  existing gallery app, I've summarized the changes requested in the
  appropriate Streambed issue and confirm that they have been applied.
- [x] If the app is on the Dash Gallery portal, I have added a link to
  the GitHub repository for the source code in the portal description.
- [x] If the app is a reimplementation of a Python gallery app for the
  DashR gallery, the app in this PR mimics, as closely as possible,
  the style and functionality of the existing app.

## The pre-review review

I have addressed all of the following questions:

- [ ] Does everything in my code serve some purpose? (I have removed
  any dead and/or irrelevant code.)
- [ ] Does everything in my code have a clear purpose? (My code is
  readable and, where it isn't, it has been commented appropriately.)]
- [ ] Am I reinventing the wheel? (I have used appropriate packages to
  lessen the volume of code that needs to be maintained.)

### CODE REVIEW

File: apps/dash-web-trader/assets/style.css [hunk 1]
```diff
@@ -0,0 +1,836 @@
...
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
+}
+
+.Select-control{
+  background-color: inherit;
+  color:#b4b4b4;
+  border-color: rgba(180, 180, 180, 0.4);
+  border-radius: 20px 20px 20px 20px / 50% 50% 50% 50%;
```
Comment (shammamah) [inline apps/dash-web-trader/assets/style.css]: Love this! It looks so cool :)

File: apps/dash-web-trader/requirements.txt [hunk 2]
```diff
@@ -0,0 +1,33 @@
+attrs==19.1.0
```
Comment (shammamah) [inline apps/dash-web-trader/requirements.txt]: Do we absolutely need all of these packages? Would be nice to trim it down to only the ones that are needed (e.g., I don't think `jupyter-core` is needed, but correct me if I'm wrong)

File: apps/dash-web-trader/app.py [hunk 3]
```diff
@@ -0,0 +1,1447 @@
...
+        className="two-col",
+        children=[
+            html.P(className="p-top-bar", children=cellTitle),
+            html.P(id=cellTitle, children=cellValue),
+        ],
+    )
+
+
+# Returns HTML Top Bar for app layout
+def get_top_bar(
```
Comment (Mskycoder) [inline apps/dash-web-trader/app.py]: When numbers get big and complicated, it gets hard to read. If it doesn't defeat the purpose of display these, might be better to trim the numbers and/or round them.

File: apps/dash-web-trader/app.py [hunk 4]
```diff
@@ -0,0 +1,1447 @@
...
+        fig = eval(study)(df, fig)
+
+    row = 1
+    # Plot trace on new row
+    for study in selected_subplots_studies:
+        row += 1
+        fig.append_trace(eval(study)(df), row, 1)
+
+    fig["layout"]["margin"] = {"t": 50, "l": 50, "b": 50, "r": 25}
+    fig["layout"]["autosize"] = True
```
Comment (Mskycoder) [inline apps/dash-web-trader/app.py]: For every refresh, the zoom on the graph gets reset.

File: apps/dash-web-trader/app.py [hunk 5]
```diff
@@ -0,0 +1,1447 @@
...
+                            )
+                        ],
+                    ),
+                ],
+            ),
+            # Chart Top Bar
+            html.Div(
+                className="row chart-top-bar",
+                children=[
+                    html.Span(
```
Comment (Mskycoder) [inline apps/dash-web-trader/app.py]: A dropdown would be a more intuitive in this case. To achieve the same look, you can change the dropdown style.

File: apps/dash-web-trader/app.py (hunk 3 above)
Comment (ronniebugia) [reply apps/dash-web-trader/app.py]: image

I feel like you lose a bunch of information when trimming the numbers. I feel like shrinking the text or something may help displaying the numbers work better.

Comment (ronniebugia) [reply apps/dash-web-trader/app.py]: I added the compressed display format but only when the number exceeds 1M

File: apps/dash-web-trader/requirements.txt [hunk 6]
```diff
@@ -1,33 +1,4 @@
-attrs==19.1.0
```
Comment (shammamah) [inline apps/dash-web-trader/requirements.txt]: Nice! The list is much shorter now :)

File: apps/dash-web-trader/app.py [hunk 7]
```diff
@@ -0,0 +1,1474 @@
...
+            html.Div(
+                className="modal-content",
+                children=[
+                    html.Span(
+                        id=pair + "closeModal", className="modal-close", children="×"
+                    ),
+                    html.P(id="modal" + pair, children=pair),
+                    # row div with two div
+                    html.Div(
+                        className="row",
```
Comment (Mskycoder) [inline apps/dash-web-trader/app.py]: The margin here is really thin, would it look better with a little spacing? 
Screen Shot 2019-07-04 at 1 04 43 PM

**Pull request merged**